src/Output.cxx

```diff
@@ -1534,6 +1534,14 @@ void ASTVisitor::GetDeclAttributes(clang::Decl const* d,
 
   if (d->hasAttr<clang::DeprecatedAttr>()) {
     attrs.push_back("deprecated");
+    clang::DeprecatedAttr* depAttr = d->getAttr<clang::DeprecatedAttr>();
+    llvm::StringRef depMsgText = "";
+    if(!depAttr->getMessage().empty())
+    {
+      depMsgText = depAttr->getMessage();
+      this->OS << " deprecation_message=\"" << depMsgText << "\"";
+    }
```

Comment:
For the `annotate` attribute we put the message directly in the `attributes=` string, e.g. `attributes="annotate(an annotation)"`, since that is how it appears in the source.  We could do the same thing here.

It looks like both attributes could use some work for escaping things inside the annotation or deprecation message string, and perhaps quoting.

Reply:
On second thought, changing the `annotation` attribute syntax (other than perhaps for XML escaping) could be a compatibility concern.

It may be better to leave the `attributes=` as-is and also add dedicated `annotation=` and `deprecation=` XML attributes to the element similar to what you propose here.  Please be sure the test case covers putting double quotes, backslashes, and angle brackets in the message.